### DISCUSSION TITLE
Question: Any real example of usage in next js

### DISCUSSION THREAD

**Discussion by Khalilsqu** [perspective-dev/perspective#2181] [Q&A]
Thanks for this amazing package!

I had a hard time installing the package in a next js project.

Could you please point me to any materials or examples (repos) of using the package in a next js project? 

Kind regards

**Comment by yangbooom** [comment]
@Khalilsqu Do you have any progress on using it in next js?
I have been working on this too, for two days.

btw, I figured out how to implement the package in Gatsby.
All you need to do is follow the Javascript example, while configuring `gatsby-node.js` file like this:

```
const PerspectivePlugin = require('@finos/perspective-webpack-plugin');

exports.onCreateWebpackConfig = ({ actions }) => {
  const { setWebpackConfig } = actions;
  setWebpackConfig({
    plugins: [new PerspectivePlugin()],
    resolve: {
      fallback: {
        stoppable: require.resolve('stoppable'),
      },
    },
  });
};
```

worth noting that stoppable package should be installed in dev dependencies.

**Comment by texodus** [comment]
We've just thrown together a minimal example of a [Next.js app with Perspective](https://github.com/ProspectiveCo/perspective-nextjs).  We'll clean it up over the next few days and add it to the docs but it works as is with `2.0.1`.

`stoppable` is a requirement of the node.js version of Perspective, not the UI, and an indication that Next.js' Webpack config is misconfigured (it is incorrectly using the SSR-driven `exports.".".node` setting from `package.json`). This means you must import Perspective's engine ESM code directly by path, rather than by package alias, for which there is no typescript declaration.  Other than that, there are no quirks with integration other than Next.js' general weirdness and insistence on owning every aspect of the build tooling - we just followed the project's directions on [custom webpack configuration](https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config).

**Reply by Ethan-Sokn** [reply]
Is this still relevant for the 3.1.4?
